### ISSUE TITLE
为什么只用YCBCR中的Y进行训练？

### ISSUE THREAD

**Issue by BarCodeReader** [twtygqyy/pytorch-vdsr#37]
我有一个疑问，那就是我看了你很多的implementation，你都是用YCBCR的Y来训练。而且你计算PSNR也都是用一个Y channel来计算的。
那么现在我有一张图，没有GT做比对，你只用Y来行进2倍或者4倍扩大，那么剩下的两个CBCR怎么办呢？双三次扩大？那么边缘对不齐啊。按照你这个思路，最后怎么才能得到RGB图呢？
为什么不直接RGB丢进去得到一个RGB的输出？

**Comment by 543877815** [comment]
双三次扩大后叠加进去

**Comment by yamengxi** [comment]
import torch
import torch.nn as nn
from math import sqrt

class Conv_ReLU_Block(nn.Module):
    def __init__(self):
        super(Conv_ReLU_Block, self).__init__()
        self.conv = nn.Conv2d(in_channels=64, out_channels=64, kernel_size=3, stride=1, padding=1, bias=False)
        self.relu = nn.ReLU(inplace=True)
        
    def forward(self, x):
        return self.relu(self.conv(x))
        
class Net(nn.Module):
    def __init__(self):
        super(Net, self).__init__()
        self.residual_layer = self.make_layer(Conv_ReLU_Block, 18)
        self.input = nn.Conv2d(in_channels=1, out_channels=64, kernel_size=3, stride=1, padding=1, bias=False)
        self.output = nn.Conv2d(in_channels=64, out_channels=1, kernel_size=3, stride=1, padding=1, bias=False)
        self.relu = nn.ReLU(inplace=True)
    
        for m in self.modules():
            if isinstance(m, nn.Conv2d):
                n = m.kernel_size[0] * m.kernel_size[1] * m.out_channels
                m.weight.data.normal_(0, sqrt(2. / n))
                
    def make_layer(self, block, num_of_layer):
        layers = []
        for _ in range(num_of_layer):
            layers.append(block())
        return nn.Sequential(*layers)

    def forward(self, x):
        residual = x
        out = self.relu(self.input(x))
        out = self.residual_layer(out)
        out = self.output(out)
        out = torch.add(out,residual)
        return out

**Comment by yamengxi** [comment]
啊
吧

**Comment by yamengxi** [comment]
A
B
